llama_stack/distribution/server/auth_providers.py

```diff
@@ -197,10 +200,22 @@ def get_attributes_from_claims(claims: dict[str, str], mapping: dict[str, str])
     return attributes
 
 
-class OAuth2TokenAuthProviderConfig(BaseModel):
+class OAuth2JWKSConfig(BaseModel):
     # The JWKS URI for collecting public keys
-    jwks_uri: str
+    uri: str
     cache_ttl: int = 3600
+
+
+class OAuth2IntrospectionConfig(BaseModel):
+    # options specific to introspection:
+    url: str
+    client_id: str
+    client_secret: str
+    send_secret_in_body: bool = False
+    tls_cafile: str | None = None
+
+
+class OAuth2TokenAuthProviderConfig(BaseModel):
```

Comment:
Do we need "Token" in the name?

Reply:
I would say no, personally, but I left the names of existing classes as they were. Happy to raise another PR for a name change (which I would think would then also rename the auth provider for consistency) if desired.

Reply:
Sounds good for a followup!